pkg/process/util/network_filter.go

```diff
@@ -0,0 +1,74 @@
+package util
+
+import (
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/stretchr/stew/slice"
+)
+
+// Connection holds a user-defined blacklisted IP, CIDR, and ports
+type Connection struct {
+	IP    string
+	CIDR  *net.IPNet
+	Ports []string
+}
+
+// takes the user defined blacklist and returns a slice of Connections
+func parseBlacklist(filters map[string][]string) (blacklist []*Connection) {
+	for ip, ports := range filters {
+		conn := &Connection{
+			IP:    ip,
+			Ports: ports,
+		}
+
+		if strings.Contains(ip, "/") {
+			ipv4, subnet, err := net.ParseCIDR(ip)
+			if err != nil {
+				log.Debugf("Could not parse %s", err)
+			}
+			conn.IP, conn.CIDR = ipv4.String(), subnet
+		}
+		blacklist = append(blacklist, conn)
+	}
+	return blacklist
+}
+
+// determine whether we should be excluding a source or destination connection
+func newNetworkFilter(direction string) (networkFilter []*Connection) {
+	if direction == "source" {
+		networkFilter = parseBlacklist(config.Datadog.GetStringMapStringSlice("system_probe_config.excluded_source_connections"))
+	} else if direction == "destination" {
+		networkFilter = parseBlacklist(config.Datadog.GetStringMapStringSlice("system_probe_config.excluded_destination_connections"))
+	}
+	return networkFilter
+}
+
+// IsBlacklistedConnection returns true if a given connection should be excluded
+// by the tracer based on user defined filters
+func IsBlacklistedConnection(dir string, addrIP Address, addrPort uint16) bool {
```

Comment:
It could be worth splitting this into two methods `IsDestinationBlacklisted` / `IsSourceBlacklisted` to avoid unexpected behavior if someone passes the wrong thing as `dir`

Reply:
Took the above comments into account and created a struct in the tracer to reference source and destination blacklists separately, while keeping this function generic.